core/services/workflows/engine.go

```diff
@@ -260,8 +260,8 @@ func (e *Engine) resumeInProgressExecutions(ctx context.Context) error {
 // and `scheduledExecution` for targets. If we don't have the necessary
 // config to initialize a scheduledExecution for a target, we'll fallback to
 // using `immediateExecution`.
-func (e *Engine) initializeExecutionStrategy(step *step) error {
-	if step.executionStrategy != nil {
+func (e *Engine) initializeExecutionStrategy(s *step) error {
```

Comment:
I find this change quite confusing, it's easy for me to get lost when there's single character vars in a large function. Can we keep it as-is?

Reply:
the linter was unhappy about shadowing. 

i could change it to `stp` or any other name than `step`. what do you suggest?

Reply:
changed and seems like the linter in my IDE is too sensitive.